File: Backend/src/main/java/com/luckyseven/backend/domain/member/service/CustomMemberDetailsService.java
package com.luckyseven.backend.domain.member.service;

import com.luckyseven.backend.domain.member.service.utill.MemberDetails;
import com.luckyseven.backend.domain.member.entity.Member;
import com.luckyseven.backend.domain.member.repository.MemberRepository;
import com.luckyseven.backend.sharedkernel.exception.CustomLogicException;
import com.luckyseven.backend.sharedkernel.exception.ExceptionCode;

import jakarta.transaction.Transactional;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Slf4j
@Service
@RequiredArgsConstructor
@Transactional
public class CustomMemberDetailsService implements UserDetailsService {

  private final MemberRepository memberRepository;


  @Override
  public MemberDetails loadUserByUsername(String email) throws UsernameNotFoundException {
    Member member = memberRepository.findByEmail(email)
        .orElseThrow(() -> new CustomLogicException(ExceptionCode.MEMBER_EMAIL_NOTFOUND,email));
    return new MemberDetails(member);
  }
  public MemberDetails loadUserById(Long id){
    Member member = memberRepository.findById(id)
        .orElseThrow(() -> new CustomLogicException(ExceptionCode.MEMBER_ID_NOTFOUND,id));
    return new MemberDetails(member);
  }

}

File: Backend/src/main/kotlin/com/luckyseven/backend/domain/member/service/CustomMemberDetailsService.kt
package com.luckyseven.backend.domain.member.service

import com.luckyseven.backend.domain.member.service.utill.MemberDetails
import com.luckyseven.backend.domain.member.entity.Member
import com.luckyseven.backend.domain.member.repository.MemberRepository
import com.luckyseven.backend.sharedkernel.exception.CustomLogicException
import com.luckyseven.backend.sharedkernel.exception.ExceptionCode
import jakarta.transaction.Transactional
import org.slf4j.LoggerFactory
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
@Transactional
class CustomMemberDetailsService(
    private val memberRepository: MemberRepository
) : UserDetailsService {
    
    private val logger = LoggerFactory.getLogger(CustomMemberDetailsService::class.java)
    
    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(email: String): MemberDetails {
        val member = memberRepository.findByEmail(email)
            .orElseThrow { CustomLogicException(ExceptionCode.MEMBER_EMAIL_NOTFOUND, email) }
        return MemberDetails(member)
    }
    
    fun loadUserById(id: Long): MemberDetails {
        val member = memberRepository.findById(id)
            .orElseThrow { CustomLogicException(ExceptionCode.MEMBER_ID_NOTFOUND, id) }
        return MemberDetails(member)
    }
} 